DEVELOPER.md

```diff
@@ -0,0 +1,131 @@
+`testapp/` contains the code for the Angular applications that e2e tests run against.
+`website/` contains code for generating Protractor API documentation and the website at protractortest.org
+
+Most of the code is written in TypeScript, with the exception of a few js files.
+
+`lib/debugger` is for element explorer, `browser.pause`, and `browser.explore`
+`lib/driverProviders` controls how WebDriver instances are created
+`lib/frameworks` contains adapters for test frameworks such as Jasmine and Mocha
+`lib/selenium-webdriver` and `lib/webdriver-js-extender` are used ONLY for API documentation generation.
+
+## Lightning Code Walkthrough
+
+TBD.
+
+## Testing
+
+Run `npm test` to run the full test suite. This assumes that you have the testapp and a
+selenium server running. Start these as separate processes with:
+
+```
+webdriver-manager update
+webdriver-manager start
+```
+
+and
+
+```
+npm start
+```
+
+This suite is described in `scripts/test.js`. It uses some small helper functions to run commands
+as child processes and capture the results, so that we can run protractor commands which should
+result in failures and verify that we get the expected number and type of failures.
+
+The suite contains unit tests, end to end tests using the built binary, and interactive tests.
+Interactive tests are for testing `browser.pause` and element explorer.
+
+End to end tests all have configuration files which live in `spec/`. Many tests do not need
+an actual Selenium server backing them, and use the `mockSelenium` configuration, which saves
+time by not connecting to a real selenium server.
+
+## Important dependencies
+
+Protractor has very close dependencies with several other projects under the Angular umbrella:
+
+`jasminewd2` is an extension of the Jasmine test framework that adds utilities for
+working with selenium-webdriver. github.com/angular/jasminewd
+
+`blocking-proxy` is a separate binary, which handles traffic between a test script and
+webdriver. It can be turned on via a protractor configuration file, and in the future
+all logic to wait for Angular will be handled through the blocking proxy.
+github.com/angular/blocking-proxy
+
+`webdriver-js-extender` extends selenium-webdriver to add Appium commands.
+github.com/angular/webdriver-js-extender
+
+## Continuous Integration
+
+PRs or changes submitted to master will automatically trigger contiuous integration on two
+different services - Travis, and Circle CI.
+
+Travis runs e2e tests via SauceLabs against a variety of browsers. The essential browsers run a
+more complete test suite. We also run a set of smoke tests against a larger set of browsers,
+which is allowed to fail. This is due to flakiness in IE, Safari and older browser versions.
+
+Circle CI runs a slightly modified version of `npm test` in a single VM. It installs
+the browsers it needs locally. Circle CI runs unit tests and a set of e2e tests against Chrome.
```

Comment:
Maybe mention why we have two systems? Circle limits you by running instance time, so it's better to keep the Saucelabs tests running on Travis. Also, stdout and stderr logs of all processes as well as a tcpdump of webdriver traffic are all available in the CircleCI artifacts for each build.